snippets/base/admin/actions.py

```diff
@@ -1,8 +1,216 @@
+            return None
+        return value
+
+    def _translate_age(value):
+        text = next(x[1] for x in PROFILE_AGE_CHOICES if x[0] == value)
+        new_value = next(x[0] for x in PROFILE_AGE_CHOICES_ASR if x[1] == text)
+        return new_value
+
+    def _translate_any_to_none(value):
+        if value == 'any':
+            return None
+        return value
+
+    def _translate_version(value):
+        value = _translate_any_to_none(value)
+        if value:
+            value = value.split('.', 1)[0]
+        return value
+
+    def _translate_is_developer(value):
+        value = _translate_any_to_none(value)
+        if value == 'yes':
+            return '=='
+        elif value == 'no':
+            return '<'
+        return value
+
+    co = snippet.client_options
+
+    addon = None
+    if co['addon_check_type'] != 'any':
+        name = co['addon_name']
+        addon, _ = Addon.objects.get_or_create(name=name,
+                                               guid=name,
+                                               url=f'https://addons.mozilla.org/{name}')
+
+    target = Target(creator=creator)
+    data = {
+        'name': f'Target for {snippet.name}',
+        'on_release': snippet.on_release,
+        'on_beta': snippet.on_beta,
+        'on_aurora': snippet.on_aurora,
+        'on_nightly': snippet.on_nightly,
+        'on_esr': snippet.on_esr,
+        'filtr_is_default_browser': _transalate_yes_no_to_true_false(co['is_default_browser']),
+        'filtr_profile_age_created_0': _translate_age(co['profileage_lower_bound']),
+        'filtr_profile_age_created_1': _translate_age(co['profileage_upper_bound']),
+        'filtr_firefox_version_0': _translate_version(co['version_lower_bound']),
+        'filtr_firefox_version_1': _translate_version(co['version_upper_bound']),
+        'filtr_previous_session_end_0': _translate_age(co['sessionage_lower_bound']),
+        'filtr_previous_session_end_1': _translate_age(co['sessionage_upper_bound']),
+        'filtr_uses_firefox_sync': _transalate_yes_no_to_true_false(co['has_fxaccount']),
+        'filtr_country': snippet.countries.all(),
+        'filtr_is_developer': _translate_is_developer(co['is_developer']),
+        'filtr_browser_addon_0': _translate_any_to_none(co['addon_check_type']),
+        'filtr_browser_addon_1': addon.id if addon else None,
+        'filtr_total_bookmarks_count_0': _translate_bookmarks(co['bookmarks_count_lower_bound']),
+        'filtr_total_bookmarks_count_1': _translate_bookmarks(co['bookmarks_count_upper_bound']),
+        # Search engine filter has not been used in AS Snippet.
+        'filtr_current_search_engine': None,
+        'client_match_rules': snippet.client_match_rules.all(),
+    }
+
+    form = TargetAdminForm(instance=target, data=data)
+    form.is_valid()
+    target = form.save()
+
+    asrsnippet.target = target
+
+
+def _migrate_locales(snippet, asrsnippet):
+    for locale in snippet.locales.all():
```

Comment:
could just do `asrsnippet.locales.add(*snippet.locales.all(), bulk=True)` to be a bit quicker, but it's not a big deal or blocker. Mostly style I think.

Reply:
I like that. You get :100: style points!

Reply:
hmm that doesn't seem work. i'll leave it as is for now.